src/main/java/org/elasticsearch/river/kafka/KafkaWorker.java

```diff
@@ -100,22 +97,10 @@ private void consumeMessagesAndAddToBulkProcessor(final KafkaStream stream) {
                 final MessageAndMetadata messageAndMetadata = consumerIterator.next();
                 logMessage(messageAndMetadata);
 
-                messageSet.add(messageAndMetadata);
```

Comment:
Hi @schonfeld,

Could you please explain why have you changed the logic and now every single message to the bulk processor. Also, you removed the counter part, and the finally block, this quite changes the default behaviour of the river.

Thanks,
Mariam